src/test/txpackage_tests.cpp

```diff
@@ -76,6 +77,35 @@ BOOST_FIXTURE_TEST_CASE(package_sanitization_tests, TestChain100Setup)
     BOOST_CHECK(!CheckPackage(package_duplicate_txids_empty, state_duplicates));
     BOOST_CHECK_EQUAL(state_duplicates.GetResult(), PackageValidationResult::PCKG_POLICY);
     BOOST_CHECK_EQUAL(state_duplicates.GetRejectReason(), "package-contains-duplicates");
+    BOOST_CHECK(!IsConsistent(package_duplicate_txids_empty));
+
+    // Packages can't have transactions spending the same prevout
+    CMutableTransaction tx_zero_1;
+    CMutableTransaction tx_zero_2;
+    COutPoint same_prevout{InsecureRand256(), 0};
+    tx_zero_1.vin.emplace_back(CTxIn{same_prevout});
+    tx_zero_2.vin.emplace_back(CTxIn{same_prevout});
+    // Different vouts (not the same tx)
+    tx_zero_1.vout.emplace_back(CTxOut(CENT, P2WSH_OP_TRUE));
+    tx_zero_2.vout.emplace_back(CTxOut(2 * CENT, P2WSH_OP_TRUE));
+    Package package_conflicts{MakeTransactionRef(tx_zero_1), MakeTransactionRef(tx_zero_2)};
+    BOOST_CHECK(!IsConsistent(package_conflicts));
+    // Transactions are considered sorted when they have no dependencies.
+    BOOST_CHECK(IsSorted(package_conflicts));
+    PackageValidationState state_conflicts;
+    BOOST_CHECK(!CheckPackage(package_conflicts, state_conflicts));
+    BOOST_CHECK_EQUAL(state_conflicts.GetResult(), PackageValidationResult::PCKG_POLICY);
+    BOOST_CHECK_EQUAL(state_conflicts.GetRejectReason(), "conflict-in-package");
+
+    // IsConsistent only cares about conflicts between transactions, not about a transaction
```

Comment:
I'm not sure what "this" you're referring to, could you be more specific? Checking that a package `IsConsistent` is just a basic sanitization check before we try to build sets of txids of prevouts. We rely heavily on the assumption that the prevouts are unique, so it's important to check that assumption.

Reply:
I meant [this code](https://github.com/bitcoin/bitcoin/blob/ad0f2336b0ae5ab132373f52e713b94dd1365f03/src/policy/packages.cpp#L68).

Adding once-at-a-time seemed more natural, no? And then you'd have an `assume not in the list already`, which would say "it's very surprising we're hitting it in `IsConsistent` because it should have been checked early".

Reply:
> I'm still slightly confused to why you want to enable this so much

This is not new code. It was added to #20833 to address [this comment](https://github.com/bitcoin/bitcoin/pull/20833#discussion_r621442515) (screenshotting because github will struggle to load it): 
![image](https://github.com/bitcoin/bitcoin/assets/25183001/88040553-77a2-45b8-87a1-30291f85b609)
 
tldr the rationale was to not change a `TX_CONSENSUS` error into a package policy error.

I agree it's an extra thing to think about when reading the code, but it's not new to this PR. This PR adds a comment and a test for this behavior because it was requested by [this comment](https://github.com/bitcoin/bitcoin/pull/26711#discussion_r1340521597).

Reply:
Thank you for clarification, i totally missed it's not new code. I agree with you now. Hopefully it doesn't distract other reviewers like i was distracted.